docs/dx.rst

```diff
@@ -0,0 +1,186 @@
+-----------------------------
+Since there are directories on DNAnexus, and  a file and a folder with the
+same path can also have duplicate names, copy and copytree work differently
+on the platform compared to other OBS services (Swift/S3).
+
+Suppose we are trying to copy a file to a destination path::
+
+    stor.copy('dx://project1:/folder/file.txt', 'dx://project2:/new_folder')
+    OR
+    Path('dx://project1:/folder/file.txt').copy('dx://project2:/new_folder')
+
+The outcome of the above commands is determined by the file/folder structure
+already present at the destination. If ``project2:/new_folder`` is an already
+existing directory, stor will set the new destination of the file as under that
+folder (i.e. as ``dx://project2:/new_folder/file.txt``). If the final destination
+path already exists, it deletes this file and copies the new file there.
+Conversely, if ``project2:/new_folder`` is not an existing directory, stor
+attempts to copy ``file.txt`` to ``project2:/new_folder`` as a file (remember:
+files without extensions are allowed, hence stor has no reason to believe
+``new_folder`` is not a filename we're trying to copy to). Thus, stor will
+copy our file to the *file* ``project2:/new_folder``. The caveat here again
+is that if ``project2:/new_folder`` already existed as a file, it is deleted,
+before a copy is attempted.
+
+Copytree also works in a similar fashion, wherein directory names are compared,
+instead of filenames. Suppose we are trying to copytree a directory::
+
+    stor.copytree('dx://project1:/folder', 'dx://project2:/new_folder')
+    OR
+    Path('dx://project1:/folder').copytree('dx://project2:/new_folder')
+
+Again, if the ``project2:/new_folder`` is an already existing directory, stor
+will set the new destination for the folder as a subfolder to that directory
+(i.e. as ``dx://project2:/new_folder/folder``. However, if this new destination is
+already present, a `TargetExistsError` is raised, instead of deleting it like in
+`DXPath.copy`, because we don't wish to delete full directories through copytree
+(use `DXPath.rmtree` for that) or merge two directories. If ``project2:/new_folder``
+is not originally present, then the folder is copied to this path.
+
+The above discussion was for DNAnexus to DNAnexus paths. The scenario shifts
+for posix to DX or DX to posix paths. For posix to DX paths, the destination
+mentioned in the copy/copytree command is never altered based on the dest
+filesystem. The original destination mentioned in the command is taken to be
+the only intended endpoint. Thus, in our scenario above, if ``project2:/new_folder``
+does not exist as a file, the copy is done to that file, else a `TargetExistsError`
+is raised. Similarly for copytree on a directory. This behavior is the same
+as other OBS services (SwiFT/S3) that has been traditionally applied.
+For DX to posix paths also, the traditional behavior has been followed,
+to maintain consistency on posix systems.
+
+Open on stor
```

Comment:
"Open on stor" section isn't particularly useful to the user - can stick this part in the OBSFile docstring or in comments on the open() method

Reply:
this comment still applies. please move this explanation to OBSFile

Reply:
Moved